.github/services/sftp/sftp/action.yml

```diff
@@ -24,7 +24,28 @@ runs:
     - name: Setup Sftp
       shell: bash
       working-directory: fixtures/sftp
-      run: docker-compose -f docker-compose-sftp.yml up -d
+      run: |
+        docker-compose -f docker-compose-sftp.yml up -d
+        set +ex
```

Comment:
I don't like this since it's hard for local development. Can we implement in docker-compose's health check instead?

Reply:
No. We use `-d` flag for `docker compose`, that's means all docker operations will be run in the background.

This is not used for local development,; people will run `docker ps` manually when the test fails. The piece of script is just for CI only.

Reply:
So we can't implement that in `health_check`?

Reply:
Yes, the health_check in `docker-compose` is been designed to help the Docker daemon detect the dead container and restart it. Not suitable for this circumstance.

Reply:
Like postgresql

```text
╭─   manjusaka@manjusaka-garuda   ~/Documents/projects/opendal   jupyter 3.12.0                                                                                                                                          manjusaka/fix-sftp-test *1   23:34:00  
╰─ docker ps
CONTAINER ID   IMAGE          COMMAND                  CREATED         STATUS                            PORTS                                           NAMES
1e70063710ea   postgres:13    "docker-entrypoint.s…"   8 seconds ago   Up 7 seconds (health: starting)   0.0.0.0:5432->5432/tcp, :::5432->5432/tcp       postgresql-postgres-db-1
729b4e400e39   mongo:latest   "docker-entrypoint.s…"   8 days ago      Up 3 hours                        0.0.0.0:27017->27017/tcp, :::27017->27017/tcp   mongo-mongo-1
```

The server still in starting status after 7s when `docker compose up -d` command finished

Reply:
Is there a standard method to confirm that the container has started following `docker compose up -d`? I prefer remove those scripts.

Reply:
Maybe we can use `docker compose up -d --wait`?

```shell
      --wait                      Wait for services to be
                                  running|healthy. Implies detached mode.
      --wait-timeout int          Maximum duration to wait for the
                                  project to be running|healthy.
```

Reply:
> that we have the same setup in both CI and locally.

We already have the same test, just need extra check for automation test in CI. 

Actually, if you want to run the test and compose by using script locally,  you need extra health check to make sure the server is running before running your test too.

Reply:
> Maybe we can use `docker compose up -d --wait`?
> 
> ```shell
>       --wait                      Wait for services to be
>                                   running|healthy. Implies detached mode.
>       --wait-timeout int          Maximum duration to wait for the
>                                   project to be running|healthy.
> ```

Seems good, I'll make a test